pkg/oam/util/helper.go

```diff
@@ -224,9 +224,15 @@ func PassLabelAndAnnotation(parentObj oam.Object, childObj labelAnnotationObject
 	childObj.SetAnnotations(MergeMap(parentObj.GetAnnotations(), childObj.GetAnnotations()))
 }
 
-// GetCRDName return the CRD name of any resources
-// the format of the CRD of a resource is <kind purals>.<group>
-func GetCRDName(u *unstructured.Unstructured) string {
+// GetDefinitionName return the Definition name of any resources
+// the format of the definition of a resource is <kind plurals>.<group>
+// Now the definition name of a resource could also be defined as `definition.oam.dev/name` in `metadata.annotations`
+func GetDefinitionName(u *unstructured.Unstructured) string {
+	if labels := u.GetAnnotations(); labels != nil {
```

Comment:
We need to get labels now, not annotations

Reply:
If tests still passed with this obviously bug, you need to write more cases I think.

Reply:
> If tests still passed with this obviously bug, you need to write more cases I think.

Sorry, the code was automatically reverted from Disk instead of Memory by Goland. My bad, fixed. Thanks.